### PR TITLE
core(cls): refactor CLS and CLS-AF

### PR DESCRIPTION
**Pull request by adamraine**
Follow up to https://github.com/GoogleChrome/lighthouse/pull/12046#discussion_r569866794 and https://github.com/GoogleChrome/lighthouse/pull/12034.

Removes duplicated code which handles the `had_recent_input` flag in layout shift events.

Plan is to merge after #12046 lands.

### CODE REVIEW

File: lighthouse-core/computed/metrics/cumulative-layout-shift-all-frames.js [hunk 1]
```diff
@@ -10,44 +10,30 @@
...
     const cumulativeShift = layoutShiftEvents
       .map(e => {
-        if (e.args.data.had_recent_input) return 0;
-
         // COMPAT: remove after m90 hits stable
         // We should replace with a LHError at that point:
         // https://github.com/GoogleChrome/lighthouse/pull/12034#discussion_r568150032
-        if (e.args.data.weighted_score_delta !== undefined) {
-          return e.args.data.weighted_score_delta;
+        if (e.weighted_score_delta === undefined) {
```
Comment (patrickhulce) [inline lighthouse-core/computed/metrics/cumulative-layout-shift-all-frames.js:31]: 👍

File: lighthouse-core/test/computed/metrics/cumulative-layout-shift-all-frames-test.js [hunk 2]
```diff
@@ -25,6 +25,27 @@ describe('Metrics: CLS All Frames', () => {
...
+      {name: 'FrameCommittedInBrowser', cat, args: {data: {frame: childFrame, parent: mainFrame, url: 'https://frame.com'}}},
+      {name: 'LayoutShift', cat, args: {frame: childFrame, data: {had_recent_input: true, score: 1, weighted_score_delta: 1, is_main_frame: false}}},
+      {name: 'LayoutShift', cat, args: {frame: childFrame, data: {had_recent_input: true, score: 1, weighted_score_delta: 1, is_main_frame: false}}},
+      {name: 'LayoutShift', cat, args: {frame: childFrame, data: {had_recent_input: false, score: 1, weighted_score_delta: 1, is_main_frame: false}}},
+      {name: 'LayoutShift', cat, args: {frame: childFrame, data: {had_recent_input: true, score: 2, weighted_score_delta: 1, is_main_frame: false}}},
+      {name: 'LayoutShift', cat, args: {frame: childFrame, data: {had_recent_input: false, score: 3, weighted_score_delta: 1, is_main_frame: false}}}
+      /* eslint-enable max-len */
+    );
+    const result = await CumulativeLayoutShiftAllFrames.request(trace, context);
+    expect(result.value).toBe(4);
```
Comment (patrickhulce) [inline lighthouse-core/test/computed/metrics/cumulative-layout-shift-all-frames-test.js:46]: move this test after "uses weighted_score_delta"? the mismatch threw me a bit :)

**Review by patrickhulce — COMMENTED** [review]
LGTM, thanks for the cleanup!

File: lighthouse-core/computed/metrics/cumulative-layout-shift.js [hunk 3]
```diff
@@ -17,57 +17,16 @@ class CumulativeLayoutShift {
...
-      };
-    }
-
-    let cumulativeLayoutShift =
-      finalLayoutShift.args &&
-      finalLayoutShift.args.data &&
-      finalLayoutShift.args.data.cumulative_score;
-
-    if (cumulativeLayoutShift === undefined) {
-      throw new LHError(LHError.errors.LAYOUT_SHIFT_MISSING_DATA);
```
[the remark this answers is not in the document]
Comment (adamraine) [reply lighthouse-core/computed/metrics/cumulative-layout-shift.js:49]: If that's the case, should we remove this error entirely?

[the remark this answers is not in the document]
Comment (adamraine) [reply lighthouse-core/computed/metrics/cumulative-layout-shift.js:49]: Update, HTTP archive didn't find anything:
```sql
CREATE TEMPORARY FUNCTION
  isFail(report STRING)
  RETURNS STRING
  LANGUAGE js AS '''
try {
  var $ = JSON.parse(report);
  var a = $.audits['cumulative-layout-shift'];
  return a.errorMessage;
} catch (e) {
  return '';
}
''';


SELECT
  item.url,
  item.report,
FROM
  `httparchive.lighthouse.2021_01_01_mobile` item
WHERE 
  isFail(item.report) = 'Something went wrong with recording the trace over your page load. Please run Lighthouse again. (LAYOUT_SHIFT_MISSING_DATA)'
LIMIT 1
```

[the remark this answers is not in the document]
Comment (patrickhulce) [reply lighthouse-core/computed/metrics/cumulative-layout-shift.js:49]: yeah let's just remove it entirely then 👍

**Review by patrickhulce — APPROVED** [review]
LGTM with the error removal 👍

**Pull request merged**